packages/fxa-settings/src/pages/Signup/ConfirmSignupCode/index.tsx

```diff
@@ -39,90 +86,199 @@ const ConfirmSignupCode = ({
+  useEffect(() => {
+    if (clearMessages) {
+      const timer = setTimeout(() => {
+        setCodeErrorMessage('');
+        setBanner({ visible: false, type: undefined, message: undefined });
+        setClearMessages(false);
+      }, 750);
+      return () => clearTimeout(timer);
+    }
+    return;
+  }, [clearMessages]);
+
+  // Hide the ResendCode button after too many attempts. Redisplay button after a delay.
+  useEffect(() => {
+    if (resendCodeCount > 3) {
+      const timer = setTimeout(() => {
+        setResendCodeCount(0);
+      }, 5000);
+      return () => clearTimeout(timer);
+    }
+    return;
+  }, [resendCodeCount]);
 
-  const onSubmit = () => {
-    if (!code) {
-      const codeRequiredError = ftlMsgResolver.getMsg(
-        'confirm-signup-code-required-error',
-        'Please enter confirmation code'
+  async function handleResendCode() {
+    try {
+      await account.sendVerificationCode();
+      setResendCodeCount(resendCodeCount + 1);
+      const localizedBannerMessage = ftlMsgResolver.getMsg(
+        'confirm-signup-code-resend-code-success-message',
+        `Email resent. Add ${accountsEmail} to your contacts to ensure a smooth delivery.`,
+        { accountsEmail }
+      );
+      setBanner({
+        visible: true,
+        type: BannerType.success,
+        message: localizedBannerMessage,
+      });
+    } catch (e) {
+      // TODO verify error message copy - This is a new error message
+      const localizedBannerMessage = ftlMsgResolver.getMsg(
+        'confirm-signup-code-error-message',
+        'Something went wrong. A new code could not be sent.'
       );
-      setCodeErrorMessage(codeRequiredError);
+      setBanner({
+        visible: true,
+        type: BannerType.error,
+        message: localizedBannerMessage,
+      });
     }
+  }
+
+  function alertSuccessAndGoForward() {
+    alertBar.success(
+      ftlMsgResolver.getMsg(
+        'confirm-signup-code-success-alert',
+        'Account confirmed successfully'
+      )
+    );
+    // TODO redirect elsewhere if relying party
+    navigate('/settings', { replace: true });
+  }
+
+  async function verifySession(code: string) {
+    logViewEvent(`flow.${viewName}`, 'submit', REACT_ENTRYPOINT);
     try {
-      // Check confirmation code
-      // Log success event (verification.success)
-      // Display success message
-      // Handle newsletter subscription
+      await account.verifySession(code);
+
+      // FOLLOW-UP: does not yet exist in Settings
+      // BEFORE: notifier.trigger('verification.success');
+      logViewEvent(
+        `flow.${viewName}`,
+        'verification.success',
+        REACT_ENTRYPOINT
+      );
+      // FOLLOW-UP: enable once state can be accessed from previous component
+      // FOLLOW-UP: notifier does not exist in Settings
+      // BEFORE: this.notifier.trigger('flow.event', {event: 'newsletter.subscribed',})
+      // if (newsletters) {
```

Comment:
Not super sure what this does under the hood either. What would the follow up look like?

Reply:
This might depend on how the newsletter subscription is handled in the `Signup` page - I'm not sure if there will be leftover tasks to handle once the user verifies their account. I assume we might not be sending newsletters to unverified emails?